PythonVisualizations/BinaryTreeBase.py

```diff
@@ -0,0 +1,515 @@
+      self.canvas.itemconfig(node.drawable.display_val, fill="black")
+
+   # draws the circle and the key value
+   def createNodeShape(self, x, y, key, tag):
+      circle = self.canvas.create_circle(x, y, self.CIRCLE_SIZE, tag = tag)
+      circle_text = self.canvas.create_text(x,y, text=key, font=self.VALUE_FONT, tag = tag)
+
+      # get rid of the none shape that was in its spot
+      items = set(self.canvas.find_overlapping(*(self.canvas.coords(circle))))
+      noneShape = set(self.canvas.find_withtag("none"))
+      noneItems = (items & noneShape)
+      
+      while len(noneItems) > 0:
+         self.canvas.delete(noneItems.pop())
+
+      self.canvas.update()
+
+      return circle, circle_text
+
+   # creates the none shape that nodes will point to
+   # line coords should be the coords of the line that points to none
+   def createNoneShape(self, node, level, lineCoords, direction, tag):
+      y = lineCoords[3] + (self.CIRCLE_SIZE // 2)
+      xCoord = self.calculateCoordinates(node, level+1, direction)[0]
+
+      shrinkFactor = self.CIRCLE_SIZE // 3
+
+      coords1 = ( xCoord-self.CIRCLE_SIZE, y, 
+                  xCoord+self.CIRCLE_SIZE, y)
+      coords2 = (  shrinkFactor + coords1[0], coords1[1] + shrinkFactor,
+                  -shrinkFactor + coords1[2], coords1[3] + shrinkFactor)
+      coords3 = (  shrinkFactor + coords2[0], coords2[1] + shrinkFactor, 
+                  -shrinkFactor + coords2[2], coords2[3] + shrinkFactor)
+
+      self.canvas.create_line(coords1, tag = (tag, "none"))
+      self.canvas.create_line(coords2, tag = (tag, "none"))
+      self.canvas.create_line(coords3, tag = (tag, "none"))
+
+   def removeNoneShapes(self, node):
+      # get the none shapes associated with the node
+      nodeShapes = set(self.canvas.find_withtag(node.tag))
+      noneShape = set(self.canvas.find_withtag("none"))
+      noneItems = (nodeShapes & noneShape)
+
+      while len(noneItems) > 0:
+         self.canvas.delete(noneItems.pop())
+
+   # delete the node's drawing
+   def removeNodeDrawing(self, node):
+      leftChild = self.getLeftChild(node)
+      rightChild =  self.getRightChild(node)
+
+      self.canvas.delete(node.drawable.display_val)
+      self.canvas.delete(node.drawable.display_shape)
+      self.canvas.delete(self.getRightLine(node))
+      self.canvas.delete(self.getLeftLine(node))
+      self.removeNoneShapes(node)
+
+   def moveNodesToPos(self, moveItems):
+      # get the coords for the node to move to
+      moveCoords = []
+      for node in moveItems:
+         node.coords = self.calculateCoordinates(self.getParent(node),self.getLevel(node), self.getChildDirection(node))
+         moveCoords.append(( node.coords[0]-self.CIRCLE_SIZE, node.coords[1]-self.CIRCLE_SIZE,
+                              node.coords[0]+self.CIRCLE_SIZE, node.coords[1]+self.CIRCLE_SIZE))
+      
+      moveItemsTags = [node.tag for node in moveItems]
+      self.moveItemsTo(moveItemsTags, moveCoords)
```

Comment:
The items passed to `moveItemsTo()` are specified by their tags, the ones created by `generateTag()`.  The tag for a node is assigned to the node's circle, the text for its value, and the lines connecting it to its children, all get the node's tag.  This allows them to be moved as a group, but there's a subtle issue.  Inside of `moveItemsTo()`, it calls `self.canvas.coords()` on each of the tags to find their current position.  The coordinates returned by that call will be for the *__lowest__* canvas item with that tag.  Currently the circle is lowest, but if we move the lines to be lower (behind) the circle, then one of them will be lowest.  The coordinates for the line are different than that of the circle, so that will change the calculation of how far to move the whole group of items with the same tag.

Reply:
Changed the tagging system in 9937d21 to deal with this problem